### ISSUE TITLE
Moving the mouse pointer slowly doesn't update the cursor position

### ISSUE THREAD

**Issue by rocifier** [adventuregamestudio/ags#416]
If I move the mouse around quickly it seems to update my cursor sprite position fairly fast. However, when moving the mouse slowly, the cursor position is not updated. The rate that this happens at is moving my mouse about 2mm per second with a relatively fast mouse pointer speed. Upon moving the mouse more quickly or stopping mouse movement entirely, the cursor sprite suddenly jumps on the screen to the location it is expected to be at.

This happens at all resolutions and background / cursor images and sizes. It happens in the tutorial game (default) scene with all cursors. It happens consistently with or without the debugger running. 

Differences between running from within AGS editor (latest) and the compiled EXE on Windows 10:

### AGS
The problem occurs quite obviously, since the cursor is completely stuck even though my physical mouse is moving

### EXE
The problem is not so obvious. The reason for this seems to be that the movement speed of the cursor is a bit slower in full-screen mode. However instead of freezing, the cursor sprite moves very slowly. It may move just a couple of centimeters on the screen by the time my mouse has moved fully across my desk.

I am using a new SteelSeries gaming mouse which works excellent in all other applications.

**Comment by ivan-mogilko** [comment]
I need to clarify something: does it happen in both windowed and fullscreen modes, or only in fullscreen?

Also, could you tell what version of AGS are you using?

**Comment by rocifier** [comment]
@ivan-mogilko It happens in both modes. I just tested fullscreen mode from the editor, and it behaved the same as the exe - when moving the mouse pointer freely, it moves fine, but when moving it slowly the cursor goes in to what I'm going to call "fine fidelity mode" where it moves particularly slowly in response to my movements. It's like the relative movement speed halves or more. When in windowed mode and moving my mouse slowly, the cursor isn't updating its position at all with slow movements.

I'm using AGS 3.4.0

**Comment by rocifier** [comment]
Looking at the mouse code, this is potentially because the dx and dy from the mouse mickeys are stored as integers. I suspect when I move the mouse slowly that these integers are just being rounded down and so no movement is added to the mouse position. This not might be 100% true though because with the quick glance I had, I couldn't see why the "real mouse position" was being saved. Also if I move the mouse out of the window and back in to another location, the mouse is set accurately to the Windows pointer location. It's possible I'm wrong and that this real location is being accurately tracked directly from dx and dy, and that the problem is with the virtual location, which I don't yet understand.

**Comment by ivan-mogilko** [comment]
A lot of code in AGS relies on Allegro 4 library, which is pretty outdated by now, so mistake or incompatibility may originate also from there.

Mouse mickeys are only tracked by AGS itself when AGS is controlling mouse movement, rather than getting its position from system. This is by default true only in fullscreen mode - that is also why I was interested if it happens in windowed mode too.

**Comment by rocifier** [comment]
I have played around with a combination of these settings but I am unable to notice any difference. One thing I have noticed which is possible at full-screen (where all movements do something) is that I can easily move the mouse in a perfect horizontal motion at slow-mid speeds even though I am moving the mouse slightly up or down while I drag it. I don't think the problem is too urgent because it is at least responsive at full-screen which is what most people seem to deploy their games at.

**Comment by rocifier** [comment]
Wow I finally figured out what the problem is. Behold, the setting that makes mouse movement awesome on large windows monitors, but wrecks AGS and DOSBox (NOT app code!):

[Imgur](https://i.imgur.com/ZtLapNF.png)

**Comment by ivan-mogilko** [comment]
Hmm, I actually have "enhanced pointer precision" ON for as long as I remember, but I guess this also combines with mouse perfomance.

**Comment by rocifier** [comment]
@ivan-mogilko do you notice any difference turning it off for AGS builds especially in windowed mode?

**Comment by ivan-mogilko** [comment]
I am sorry, I intended to reply much earlier, but forgot about this.

> @ivan-mogilko do you notice any difference turning it off for AGS builds especially in windowed mode?

There is a significant difference between "enhanced precision" mode on and off in general (in Windows), with "off" cursor movements are slower, and since I got used to this mode it feels like mouse became much "heavier" without it.

But important thing is that I cannot see much difference between cursor movement in AGS games and desktop (or other programs).
